Source/Extensions/Blazorise.DataGrid/_DataGridMultiSelectAll.razor.cs

```diff
@@ -19,23 +19,9 @@ internal Task IsCheckedChanged( bool e )
 
         protected override Task OnParametersSetAsync()
         {
-            var hasSelectedRows = ParentDataGrid.SelectedRows?.Any() ?? false;
-
-            if ( hasSelectedRows )
-            {
-                var hasData = ParentDataGrid.DisplayData.Any();
-                var unselectedRows = ParentDataGrid.DisplayData.Except( ParentDataGrid.SelectedRows ).Any();
-
-                if ( hasSelectedRows && !unselectedRows && hasData )
-                    IsChecked = true;
-
-                if ( !hasSelectedRows || unselectedRows )
-                    IsChecked = false;
-            }
-            else
-            {
-                IsChecked = false;
-            }
+            IsChecked = ( ParentDataGrid.SelectedRows?.Any() ?? false )

```

Comment:
Is this better?

Reply:
Haha I was just talking about the flow of the feature in general. Not trimming down code. 
You can leave the new one, if the logic is exactly the same. :P

Reply:
Haha well, it doesn't hurt.

In case I'm not missing something I think the logic stays the same.